### PR TITLE
feat(Android, Stack v5): add support for disabling menu items

### PR DESCRIPTION
**Pull request by kligarski**
## Description

Adds support for disabling menu items in toolbar menu.

Closes https://github.com/software-mansion/react-native-screens-labs/issues/1497

## Changes

- add support on JS/native side for `disabled` prop
- update `test-stack-toolbar-menu-icon-android` SFT to include testing disabled item's icon tint
- add `test-stack-toolbar-menu-disabled-android` SFT to test disabling menu item in toolbar, in overflow menu, a submenu and submenu item

## Visual documentation

https://github.com/user-attachments/assets/1d0aa842-3725-4c8c-94f2-95f86ceb78d2

## Test plan

Run `test-stack-toolbar-menu-icon-android`, `test-stack-toolbar-menu-disabled-android` SFTs.

## Checklist

- [x] Included code example that can be used to test this change.
- [x] For visual changes, included screenshots / GIFs / recordings documenting the change.
- [x] For API changes, updated relevant public types.
- [x] Ensured that CI passes

### CODE REVIEW

File: src/components/gamma/stack/header/StackHeaderConfig.android.types.ts [hunk 1]
```diff
@@ -180,11 +187,9 @@ export interface StackHeaderToolbarMenuItemBaseAndroid {
    * it is disabled.
    *
    * @remarks
-   * Disabling menu elements isn't currently supported.
-   *
-   * Due to native platform limitations, if you set this prop, you must also
+   * Due to native platform limitations, if you set this prop, you should also
    * provide `iconTintColorNormal`. Otherwise, the icon will become
-   * transparent.
+   * transparent when the item is not disabled.
```
[the remark this answers is not in the document]
Comment (kligarski) [reply src/components/gamma/stack/header/StackHeaderConfig.android.types.ts:192]: I think it's okay here to emphasize that this is the only state in which the icon will have the tint but I'm open to changing it.

File: apps/src/tests/single-feature-tests/stack-v5/test-stack-toolbar-menu-disabled-android/scenario.md [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+
+**Description:** Tests the `disabled` prop on Android toolbar menu items. Covers
+all disable-relevant element kinds — an action button shown in the toolbar
+(`showAsAction: always`), an action item in the overflow menu (`showAsAction:
+never`), checkable items in a selection group, and a submenu — across both
+flows: the props flow (via the `toolbarMenu` prop) and the imperative command
+flow (via `setToolbarMenuItemOptions`). A disabled element is greyed out/correct
+tint is applied and ignores interaction: it emits neither `onPress` nor
+`onSelectionChange`, a disabled submenu cannot be opened, and a disabled checked
+item keeps its checked state.
```
[the remark this answers is not in the document]
Comment (kligarski) [reply apps/src/tests/single-feature-tests/stack-v5/test-stack-toolbar-menu-disabled-android/scenario.md:13]: For me it's okay (it is grayed out OR correct tint is applied, e.g. for icon in toolbar), let me know if I should change it.

**Review by t0maboro — APPROVED** [review]
